app/src/components/content/ContentTile.spec.ts

```diff
@@ -37,4 +47,32 @@ describe("ContentTile", () => {
 
         expect(wrapper.text()).toContain("Jan 1, 2024");
     });
+
+    it("navigates to the correct route on click", async () => {
+        const routes = [{ path: "/:slug", name: "post", component: SingleContent }];
+
+        const router = createRouter({
+            history: createWebHistory(),
+            routes,
+        });
+
+        const wrapper = mount(ContentTile, {
+            props: {
+                content: mockEnglishContentDto,
+            },
+            global: {
+                plugins: [router],
+            },
+        });
+
+        const push = vi.spyOn(wrapper.vm.$router, "push");
+
+        await wrapper.trigger("click");
+
+        await router.push(`/${mockEnglishContentDto.slug}`);
```

Comment:
It feels like we are simulating what the component itself should do here. The component should have called the route, and we must check if the router received the call, but now the test is also pushing the route...

Reply:
To expand on this, it shouldn't be needed to create a router and routes using `createRouter`, as that gives very limited value in our tests, they're not testing the actual code. Instead, you can only mock the router `push`, and then test that. Something like (copied from `ProfileMenu.spec.ts`):
```ts
const routePushMock = vi.hoisted(() => vi.fn());
vi.mock("vue-router", () => ({
    useRouter: vi.fn().mockImplementation(() => ({
        push: routePushMock,
    })),
}));
```

and then the assertion would be something like

```ts
expect(routePushMock).toHaveBeenCalledWith({route: 'post'})
```